Solutions/Azure Activity/Analytic Rules/TimeSeriesAnomaly_Mass_Cloud_Resource_Deletions.yaml

```diff
@@ -24,33 +21,31 @@ tags:
 query: |
   let starttime = 14d;
   let endtime = 1d;
-  let timeframe = 1h;
-  let TotalEventsThreshold = 25; 
-  let TimeSeriesData = 
-  AzureActivity 
-  | where TimeGenerated between (startofday(ago(starttime))..startofday(ago(endtime)))
+  let timeframe = 1d;
+  let TotalEventsThreshold = 25;
+  let TimeSeriesData = AzureActivity 
+  | where TimeGenerated between (startofday(ago(starttime))..startofday(now())) 
   | where OperationNameValue endswith "delete" 
   | project TimeGenerated, Caller 
-  | make-series Total = count() on TimeGenerated from startofday(ago(starttime)) to startofday(ago(endtime)) step timeframe by Caller; 
-  let TimeSeriesAlerts = materialize(TimeSeriesData 
+  | make-series Total = count() on TimeGenerated from startofday(ago(starttime)) to startofday(now()) step timeframe by Caller;
+  TimeSeriesData 
   | extend (anomalies, score, baseline) = series_decompose_anomalies(Total, 3, -1, 'linefit') 
   | mv-expand Total to typeof(double), TimeGenerated to typeof(datetime), anomalies to typeof(double), score to typeof(double), baseline to typeof(long) 
+  | where TimeGenerated >= startofday(ago(endtime)) 
   | where anomalies > 0 
   | project Caller, TimeGenerated, Total, baseline, anomalies, score 
-  | where Total > TotalEventsThreshold and baseline > 0 ); 
-  TimeSeriesAlerts 
-  | where TimeGenerated > (ago(endtime)) 
-  | project TimeGenerated, Caller 
+  | where Total > TotalEventsThreshold and baseline > 0 
   | join (AzureActivity 
-  | where TimeGenerated > (ago(endtime)) 
+  | where TimeGenerated > startofday(ago(endtime)) 
   | where OperationNameValue endswith "delete" 
-  | summarize count(), make_set(OperationNameValue), make_set(Resource) by bin(TimeGenerated, 1h), Caller) on TimeGenerated, Caller 
-  | extend timestamp = TimeGenerated, AccountCustomEntity = Caller
-
+  | summarize count(), make_set(OperationNameValue), make_set(_ResourceId) by bin(TimeGenerated, timeframe), Caller ) on TimeGenerated, Caller 
+  | extend Name = tostring(split(Caller,'@',0)[0]), UPNSuffix = tostring(split(Caller,'@',1)[0])
```

Comment:
Might not be a UPN - could be an SP ID - we need to account for this

Reply:
same

Reply:
done